custom_components/tahoma/__init__.py

```diff
@@ -6,21 +6,71 @@
 
 from pyhoma.client import TahomaClient
 from pyhoma.exceptions import BadCredentialsException, TooManyRequestsException
+import voluptuous as vol
 
+from homeassistant import config_entries
 from homeassistant.components.scene import DOMAIN as SCENE
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP
+from homeassistant.const import (
+    CONF_EXCLUDE,
+    CONF_PASSWORD,
+    CONF_USERNAME,
+    EVENT_HOMEASSISTANT_STOP,
+)
 from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
 
 from .const import DOMAIN, TAHOMA_TYPES
 from .coordinator import TahomaDataUpdateCoordinator
 
 _LOGGER = logging.getLogger(__name__)
 DEFAULT_UPDATE_INTERVAL = timedelta(seconds=10)
 
+CONFIG_SCHEMA = vol.Schema(
+    {
+        DOMAIN: vol.Schema(
+            {
+                vol.Required(CONF_USERNAME): cv.string,
+                vol.Required(CONF_PASSWORD): cv.string,
+                cv.deprecated(CONF_EXCLUDE, default=[]): vol.All(
```

Comment:
I deprecated CONF_EXCLUDE, since disabling entities is supported in HA by default.

Reply:
What does it mean for the end user? What occurs when it set this option?

Reply:
```2020-08-02 22:11:31 WARNING (MainThread) [custom_components.tahoma] The 'exclude' option is deprecated, please remove it from your configuration```

However, I now face an error in CONFIG_SCHEMA after I added the following:
```
  exclude: [BridgeHUEComponent, HueLampHUEComponent, PodComponent]
```

```
2020-08-02 22:11:31 ERROR (MainThread) [homeassistant.config] Unknown error calling tahoma CONFIG_SCHEMA
Traceback (most recent call last):
  File "/Users/mick/Projects/personal-projects/home-assistant-core/homeassistant/config.py", line 759, in async_process_component_config
    return component.CONFIG_SCHEMA(config)  # type: ignore
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 386, in validate_mapping
    cval = cvalue(key_path, value)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 272, in __call__
    return self._compiled([], data)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 594, in validate_dict
    return base_validate(path, iteritems(data), out)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 373, in validate_mapping
    new_key = ckey(key_path, key)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/venv/lib/python3.8/site-packages/voluptuous/schema_builder.py", line 817, in validate_callable
    return schema(data)
  File "/Users/mick/Projects/personal-projects/home-assistant-core/homeassistant/helpers/config_validation.py", line 717, in validator
    value = config[key]
TypeError: string indices must be integers
```

Reply:
That's weird, because this is the original config structure.

Reply:
Fixed. Apparently the cv.deprecated should be somewhere else, haha.